{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "<section style=\"color: #fff;\n",
    "    text-align: center;\n",
    "    background-color: #337ab7;\n",
    "    background-image: linear-gradient(120deg, #155799, #337ab7, #155799);\n",
    "    padding: 1.5rem 2rem;\n",
    "    margin: 0px 0 20px;\n",
    "    border-bottom: 1px solid #eee;\">\n",
    "  <h1 class=\"title toc-ignore project-name\" style=\"background-color:transparent;\">Les principes FAIR appliqués à la bioinformatique</h1>\n",
    "  <h2 class=\"subtitle project-tagline\">Outils et environnements FAIR pour la bioanalyse</h3>\n",
    "  <h3 class=\"subtitle project-tagline\">Thomas Denecker - <i>Data brokering</i> - IFB core</h4> \n",
    "  \n",
    "<div id=\"banner\">\n",
    "    <div style=\"display: inline-block;\">\n",
    "        <a href=\"https://twitter.com/DeneckerThomas\" target=\"_blank\"> <img src='https://upload.wikimedia.org/wikipedia/fr/c/c8/Twitter_Bird.svg' width='30px' style='margin: 10px 10px 0px 10px;'> </a>\n",
    "    </div>\n",
    "    <div style=\"display: inline-block;\">\n",
    "        <a href=\"mailto:thomas.denecker@france-bioinformatique.fr\" target=\"_blank\"> <img src='https://upload.wikimedia.org/wikipedia/fr/a/a7/Mail_%28Apple%29_logo.png' width='30px' style='margin: 10px 10px 0px 10px;'> </a>\n",
    "    </div>\n",
    "    <div style=\"display: inline-block;\">\n",
    "        <a href=\"https://github.com/thomasdenecker\" target=\"_blank\"> <img src='https://github.githubassets.com/images/modules/logos_page/Octocat.png' width='30px' style='margin: 10px 10px 0px 10px;' > </a>\n",
    "    </div>\n",
    "</div>\n",
    "</section>\n",
    "\n",
    "## Rejouer son article en un click\n",
    "\n",
    "### Cas d'étude \n",
    "\n",
    "#### Contexte\n",
    "L'objectif de l'étude est d'étudier la réponse à une privation en fer chez l'algue verte *Ostreococcus tauri* [Lelandais et al, 2016](https://doi.org/10.1186/s12864-016-2666-6). Il s'agit d'un organisme de 13.0328 Mb séparé en 20 chromosomes.\n",
    "\n",
    "L'expérience est composée de 16 échantillons de données RNAseq (triplicat, single-end de 100bp). Le plan d'expérience est illustré ci-dessous. Pour accompagner cette démonstration, nous avons choisi les échantillons S11 et S12 (reponse adaptative à long terme,)\n",
    "\n",
    "![Figure 1](https://github.com/IFB-ElixirFr/journeeAxeBioinfo_2020/raw/main/img/planExperimental.png)\n",
    "\n",
    "**Figure 1** | Plan expérimental \n",
    "\n",
    "#### Données utilisées\n",
    "- Sequence: [GCF_000214015.3_version_140606_genomic.fna](https://www.ncbi.nlm.nih.gov/assembly/GCF_000214015.3/)\n",
    "- Annotation: GCF_000214015.3_version_140606_genomic.gff\n",
    "- Données RNAseq : des [données réuduites](https://doi.org/10.5281/zenodo.3997237) au chromosome 18 (pour la démo). Le projet avec les données non réduites est disponible sur [SRA](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR4026187).\n",
    "\n",
    "#### Analyse de données \n",
    "\n",
    "Le but de l'analyse de données est de mettre en évidence une liste de gènes différentiellement exprimés lorsque que le milieu de culture à une concetration en fer plus faible que dans un milieu standard. Les différentes étapes sont présentées dans la figure suivante : \n",
    "\n",
    "![Figure 2](https://github.com/IFB-ElixirFr/journeeAxeBioinfo_2020/raw/main/img/DataAnalysis.png)\n",
    "\n",
    "**Figure 2** | Étape de l'analyse de données \n",
    "\n",
    "## Comment rejouer toute l'analyse en un click ? \n",
    "\n",
    "Un reviewer souhaite refaire vos analyses ? Vous souhaitez analyser de nouvelles données exactement de la même façon que précédement ? \n",
    "\n",
    "Si c'est FAIR, c'est sans soucis ! Vous n'êtes pas convaincu ? Une démo ? \n",
    "\n",
    "<img style=\"margin: 0px auto;\" src=\"https://github.com/IFB-ElixirFr/journeeAxeBioinfo_2020/raw/main/img/connexion.jpg\" width=\"400\"> \n",
    "\n",
    "### Étape 1 - Identification à Jupyter Hub\n",
    "\n",
    "![Figure 3](https://github.com/IFB-ElixirFr/journeeAxeBioinfo_2020/raw/main/img/JourneeAxeStep_1.png)\n",
    "\n",
    "### Étape 2 - Création d'un job Jupyter Lab\n",
    "\n",
    "![Figure 4](https://github.com/IFB-ElixirFr/journeeAxeBioinfo_2020/raw/main/img/JourneeAxeStep_2.png)\n",
    "\n",
    "### Étape 3 - Création / ouverture d'un notebook Jupyter\n",
    "\n",
    "![Figure 5](https://github.com/IFB-ElixirFr/journeeAxeBioinfo_2020/raw/main/img/JourneeAxeStep_3.png)\n",
    "\n",
    "### Étape 4 - Import du code et des données \n",
    "\n",
    "![Figure 5](https://github.com/IFB-ElixirFr/journeeAxeBioinfo_2020/raw/main/img/JourneeAxeStep_4.png)\n",
    "\n",
    "#### Création d'un répertoire de travail"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir demo\n",
    "cd demo "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Import des données\n",
    "\n",
    "**Téléchargement des données**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "wget https://zenodo.org/record/3997237/files/FAIR_Bioinfo_data.tar.gz?download=1 -O FAIR_Bioinfo_data.tar.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Extraction des données**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tar -xvzf FAIR_Bioinfo_data.tar.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Récupération des scripts d'analyse\n",
    "\n",
    "**Snakemake files - Téléchargement**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "wget https://raw.githubusercontent.com/IFB-ElixirFr/journeeAxeBioinfo_2020/main/config.yml\n",
    "wget https://raw.githubusercontent.com/IFB-ElixirFr/journeeAxeBioinfo_2020/main/demo.smk"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Snakemake files - Contenu**\n",
    "\n",
    "``` Python\n",
    "SAMPLES, = glob_wildcards(config[\"dataDir\"]+\"{sample}.fastq.gz\")\n",
    "BIDX = [\"1\",\"2\",\"3\",\"4\",\"rev.1\",\"rev.2\"]\n",
    "\n",
    "rule all:\n",
    "  input:\n",
    "    expand(\"FastQC/{sample}_fastqc.html\", sample=SAMPLES),\n",
    "    expand(\"Tmp/Otauri.{ext}.bt2\", ext=BIDX),\n",
    "    expand(\"Tmp/{sample}.sam\", sample=SAMPLES),\n",
    "    expand(\"Result/{sample}_sort.bam.bai\", sample=SAMPLES),\n",
    "    expand(\"Result/Counts/{sample}_ftc.txt\", sample=SAMPLES),\n",
    "    \"Result/counts_matrix.txt\"\n",
    "\n",
    "rule matrix_counts:\n",
    "  output:\n",
    "    \"Result/counts_matrix.txt\"\n",
    "  input:\n",
    "    countfile=expand(\"Tmp/{sample}_ftc7.txt\", sample=SAMPLES),\n",
    "    geneID=expand(\"Tmp/{sample}_ftc1.txt\", sample=SAMPLES)\n",
    "  log:\n",
    "    \"Logs/matrix_counts.log\"\n",
    "  shell:\n",
    "    \"\"\"cp {input.geneID[0]} Tmp/ftc_geneID.txt ; paste Tmp/ftc_geneID.txt {input.countfile} > {output}\"\"\"\n",
    "\n",
    "rule extract_counts:\n",
    "  output:\n",
    "    col7=\"Tmp/{sample}_ftc7.txt\",\n",
    "    col1=\"Tmp/{sample}_ftc1.txt\"\n",
    "  input:\n",
    "    \"Result/Counts/{sample}_ftc.txt\"\n",
    "  shell: \"\"\"cut -f 7 {input} | sed 1d > {output.col7} ; cut -f 1 {input} | sed 1d > {output.col1} \"\"\"\n",
    "\n",
    "rule counting:\n",
    "  output:\n",
    "    \"Result/Counts/{sample}_ftc.txt\"\n",
    "  input:\n",
    "    bam=\"Result/{sample}_sort.bam\",\n",
    "    annot=config[\"dataDir\"]+config[\"annots\"]\n",
    "  params: t=\"gene\", g=\"ID\", s=\"2\"\n",
    "  log:\n",
    "    \"Logs/{sample}_counts.log\"\n",
    "  conda: \"condaEnv4SmkRules/counting.yml\"\n",
    "  shell:\n",
    "    \"featureCounts -t {params.t} -g {params.g} -a {input.annot} -s {params.s} -o {output} {input.bam} &> {log}\"\n",
    "\n",
    "rule sam2bam_sort:\n",
    "  output:\n",
    "    bam=\"Result/{sample}_sort.bam\",\n",
    "    bai=\"Result/{sample}_sort.bam.bai\"\n",
    "  input:\n",
    "    \"Tmp/{sample}.sam\"\n",
    "  log:\n",
    "    sort=\"Logs/{sample}_sam2bam_sort.log\",\n",
    "    index=\"Logs/{sample}_bam2bai.log\"\n",
    "  conda: \"condaEnv4SmkRules/sam2bam_sort.yml\"\n",
    "  shell: \n",
    "    \"samtools sort -O bam -o {output.bam} {input} 2> {log.sort} ;\"\n",
    "    \"samtools index {output.bam} 2> {log.index}\"\n",
    "\n",
    "\n",
    "rule bwt2_mapping:\n",
    "  output:\n",
    "    \"Tmp/{sample}.sam\"\n",
    "  input:\n",
    "    config[\"dataDir\"]+\"{sample}.fastq.gz\",\n",
    "    expand(\"Tmp/Otauri.{ext}.bt2\", ext=BIDX)\n",
    "  log:\n",
    "    \"Logs/{sample}_bwt2_mapping.log\"\n",
    "  conda: \"condaEnv4SmkRules/bwt2_mapping.yml\"\n",
    "  shell: \"bowtie2 -x Tmp/Otauri -U {input[0]} -S {output} 2> {log} \"\n",
    "\n",
    "rule genome_bwt2_index:\n",
    "  output:\n",
    "    expand(\"Tmp/Otauri.{ext}.bt2\", ext=BIDX)\n",
    "  input:\n",
    "    config[\"dataDir\"]+config[\"genome\"]\n",
    "  log:\n",
    "    log1=\"Logs/genome_bwt2_index.log1\",\n",
    "    log2=\"Logs/genome_bwt2_index.log2\"\n",
    "  conda: \"condaEnv4SmkRules/bwt2_mapping.yml\"\n",
    "  shell: \"bowtie2-build {input} Tmp/Otauri 1>{log.log1} 2>{log.log2}\"\n",
    "\n",
    "rule fastqc:\n",
    "  output: \n",
    "    \"FastQC/{sample}_fastqc.zip\",\n",
    "    \"FastQC/{sample}_fastqc.html\"\n",
    "  input: \n",
    "    config[\"dataDir\"]+\"{sample}.fastq.gz\"\n",
    "  log:\n",
    "    log1=\"Logs/{sample}_fastqc.log1\",\n",
    "    log2=\"Logs/{sample}_fastqc.log2\"\n",
    "  conda: \"condaEnv4SmkRules/fastqc.yml\"\n",
    "  shell: \"fastqc --outdir FastQC/ {input} 1>{log.log1} 2>{log.log2}\"\n",
    "```\n",
    "\n",
    "**Contenu de config.yml**\n",
    "\n",
    "``` YAML\n",
    "genome:\n",
    "  O.tauri_genome.fna\n",
    "annots:\n",
    "  O.tauri_annotation.gff\n",
    "dataDir:\n",
    "  Data/\n",
    "channels:\n",
    "  - conda-forge\n",
    "  - bioconda\n",
    "  - main\n",
    "  - r\n",
    "  - default\n",
    "dependencies:\n",
    "  # Specify python version (not required but can help with downstream conflicts)\n",
    "  - python=3.7.6\n",
    "  # The workflow manager\n",
    "  - snakemake-minimal=5.10.0\n",
    "  # For visualizing workflows\n",
    "  - graphviz=2.42.3\n",
    "  - xorg-libxrender\n",
    "  - xorg-libxpm\n",
    "  # For downloading files\n",
    "  - wget=1.20.1\n",
    "  # For the RNAseq analysis\n",
    "  - fastqc=0.11.9\n",
    "  - bowtie2=2.4.1\n",
    "  - samtools=1.10\n",
    "  - subread=2.0.1\n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Slurm file - Téléchargement**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "wget https://raw.githubusercontent.com/IFB-ElixirFr/journeeAxeBioinfo_2020/main/runAnalysis.slurm"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Slurm file - Contenu**\n",
    "\n",
    "``` bash\n",
    "#!/bin/bash\n",
    "#\n",
    "#SBATCH --mem 50GB\n",
    "#SBATCH --cpus-per-task 30\n",
    "#SBATCH --partition fast\n",
    "\n",
    "module load snakemake fastqc bowtie2 samtools subread slurm-drmaa\n",
    "\n",
    "snakemake --drmaa --jobs=$SLURM_CPUS_PER_TASK -s demo.smk --configfile config.yml\n",
    "\n",
    "singularity exec -B $PWD:/home/ journee_axe_bioinfo_latest.sif Rscript --vanilla sartools.R\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Fichers pour l'analyse différentielle - Téléchargement**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "wget https://raw.githubusercontent.com/IFB-ElixirFr/journeeAxeBioinfo_2020/main/condition.txt\n",
    "wget https://raw.githubusercontent.com/IFB-ElixirFr/journeeAxeBioinfo_2020/main/sartools.R"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Contenu du script R** \n",
    "\n",
    "``` R\n",
    "library(SARTools)\n",
    "\n",
    "################################################################################\n",
    "### R script to compare several conditions with the SARTools and DESeq2 packages\n",
    "### Hugo Varet\n",
    "### November 28th, 2019\n",
    "### designed to be executed with SARTools 1.7.3\n",
    "################################################################################\n",
    "\n",
    "################################################################################\n",
    "###                parameters: to be modified by the user                    ###\n",
    "################################################################################\n",
    "rm(list=ls())                                        # remove all the objects from the R session\n",
    "\n",
    "workDir <- \"/home/\"      # working directory for the R session\n",
    "\n",
    "projectName <- \"Demo\"                         # name of the project\n",
    "author <- \"Thomas Denecker\"                                # author of the statistical analysis/report\n",
    "\n",
    "targetFile <- \"condition.txt\"                           # path to the design/target file\n",
    "rawDir <- \"Result/Counts\"                                      # path to the directory containing raw counts files\n",
    "featuresToRemove <- c(\"alignment_not_unique\",        # names of the features to be removed\n",
    "                      \"ambiguous\", \"no_feature\",     # (specific HTSeq-count information and rRNA for example)\n",
    "                      \"not_aligned\", \"too_low_aQual\")# NULL if no feature to remove\n",
    "\n",
    "varInt <- \"Iron\"                                    # factor of interest\n",
    "condRef <- \"STANDARD\"                                      # reference biological condition\n",
    "batch <- NULL                                        # blocking factor: NULL (default) or \"batch\" for example\n",
    "\n",
    "fitType <- \"parametric\"                              # mean-variance relationship: \"parametric\" (default), \"local\" or \"mean\"\n",
    "cooksCutoff <- TRUE                                  # TRUE/FALSE to perform the outliers detection (default is TRUE)\n",
    "independentFiltering <- TRUE                         # TRUE/FALSE to perform independent filtering (default is TRUE)\n",
    "alpha <- 0.05                                        # threshold of statistical significance\n",
    "pAdjustMethod <- \"BH\"                                # p-value adjustment method: \"BH\" (default) or \"BY\"\n",
    "\n",
    "typeTrans <- \"VST\"                                   # transformation for PCA/clustering: \"VST\" or \"rlog\"\n",
    "locfunc <- \"median\"                                  # \"median\" (default) or \"shorth\" to estimate the size factors\n",
    "\n",
    "colors <- c(\"#f3c300\", \"#875692\", \"#f38400\",         # vector of colors of each biological condition on the plots\n",
    "            \"#a1caf1\", \"#be0032\", \"#c2b280\",\n",
    "            \"#848482\", \"#008856\", \"#e68fac\",\n",
    "            \"#0067a5\", \"#f99379\", \"#604e97\")\n",
    "\n",
    "forceCairoGraph <- FALSE\n",
    "\n",
    "################################################################################\n",
    "###                             running script                               ###\n",
    "################################################################################\n",
    "setwd(workDir)\n",
    "library(SARTools)\n",
    "if (forceCairoGraph) options(bitmapType=\"cairo\")\n",
    "\n",
    "# checking parameters\n",
    "checkParameters.DESeq2(projectName=projectName,author=author,targetFile=targetFile,\n",
    "                       rawDir=rawDir,featuresToRemove=featuresToRemove,varInt=varInt,\n",
    "                       condRef=condRef,batch=batch,fitType=fitType,cooksCutoff=cooksCutoff,\n",
    "                       independentFiltering=independentFiltering,alpha=alpha,pAdjustMethod=pAdjustMethod,\n",
    "                       typeTrans=typeTrans,locfunc=locfunc,colors=colors)\n",
    "\n",
    "# loading target file\n",
    "target <- loadTargetFile(targetFile=targetFile, varInt=varInt, condRef=condRef, batch=batch)\n",
    "\n",
    "# loading counts\n",
    "counts <- loadCountData(target=target, rawDir=rawDir, featuresToRemove=featuresToRemove)\n",
    "\n",
    "# description plots\n",
    "majSequences <- descriptionPlots(counts=counts, group=target[,varInt], col=colors)\n",
    "\n",
    "# analysis with DESeq2\n",
    "out.DESeq2 <- run.DESeq2(counts=counts, target=target, varInt=varInt, batch=batch,\n",
    "                         locfunc=locfunc, fitType=fitType, pAdjustMethod=pAdjustMethod,\n",
    "                         cooksCutoff=cooksCutoff, independentFiltering=independentFiltering, alpha=alpha)\n",
    "\n",
    "# PCA + clustering\n",
    "exploreCounts(object=out.DESeq2$dds, group=target[,varInt], typeTrans=typeTrans, col=colors)\n",
    "\n",
    "# summary of the analysis (boxplots, dispersions, diag size factors, export table, nDiffTotal, histograms, MA plot)\n",
    "summaryResults <- summarizeResults.DESeq2(out.DESeq2, group=target[,varInt], col=colors,\n",
    "                                          independentFiltering=independentFiltering,\n",
    "                                          cooksCutoff=cooksCutoff, alpha=alpha)\n",
    "\n",
    "# save image of the R session\n",
    "save.image(file=paste0(projectName, \".RData\"))\n",
    "\n",
    "# generating HTML report\n",
    "writeReport.DESeq2(target=target, counts=counts, out.DESeq2=out.DESeq2, summaryResults=summaryResults,\n",
    "                   majSequences=majSequences, workDir=workDir, projectName=projectName, author=author,\n",
    "                   targetFile=targetFile, rawDir=rawDir, featuresToRemove=featuresToRemove, varInt=varInt,\n",
    "                   condRef=condRef, batch=batch, fitType=fitType, cooksCutoff=cooksCutoff,\n",
    "                   independentFiltering=independentFiltering, alpha=alpha, pAdjustMethod=pAdjustMethod,\n",
    "                   typeTrans=typeTrans, locfunc=locfunc, colors=colors)\n",
    "```\n",
    "\n",
    "**Contenu du fichier contition.txt**\n",
    "\n",
    "```\n",
    "label\tfiles\tIron\n",
    "D1\tSRR3105699_chr18_ftc.txt\tDEPLETED\n",
    "D2\tSRR3105698_chr18_ftc.txt\tDEPLETED\n",
    "D3\tSRR3105697_chr18_ftc.txt\tDEPLETED\n",
    "S1\tSRR3099587_chr18_ftc.txt\tSTANDARD\n",
    "S2\tSRR3099586_chr18_ftc.txt\tSTANDARD\n",
    "S3\tSRR3099585_chr18_ftc.txt\tSTANDARD\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Récupération de l'environnement pour l'analyse différentielle\n",
    "\n",
    "#### Docker \n",
    "\n",
    "> Pourquoi ne pas faire dans conda comme précédement  ? \n",
    "\n",
    "Un des défauts de R est qu'il n'est pas possible simplement de récupérer une version d'un package. Une solution est donc de figé R et ses packages lors de l'analyse dans une image docker. \n",
    "\n",
    "![Figure 6](https://github.com/IFB-ElixirFr/journeeAxeBioinfo_2020/raw/main/img/docker_singularity.png)\n",
    "\n",
    "> Comment as tu construit l'image docker utilisée dans cette analyse ? \n",
    "\n",
    "1) Écrire un Dockerfile\n",
    "\n",
    "``` bash\n",
    "FROM rocker/tidyverse\n",
    "MAINTAINER Thomas DENECKER (thomas.denecker@france-bioinformatique.fr)\n",
    "RUN Rscript -e 'devtools::install_github(\"PF2-pasteur-fr/SARTools\", build_opts=\"--no-resave-data\")'\n",
    "```\n",
    "2) Construire à l'aide de `docker build` \n",
    "\n",
    "``` bash\n",
    "docker build -t tdenecker/journee_axe_bioinfo .\n",
    "```\n",
    "\n",
    "> Comment as tu partagé cette image ?\n",
    "\n",
    "``` bash\n",
    "docker push tdenecker/journee_axe_bioinfo\n",
    "```\n",
    "\n",
    "#### Utiliser une image docker sur le cluster via Singularity\n",
    "\n",
    "Notamment pour des questions de sécurité, il n'est pas possible d'utiliser directement docker sur le cluster de l'IFB mais singularity. Heureusement, il est très simple de construire une image singularity à partir de l'image docker."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "SINGULARITY_LOCALCACHEDIR=${PWD}\n",
    "SINGULARITY_PULLFOLDER=${PWD}\n",
    "singularity pull docker://tdenecker/journee_axe_bioinfo"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Étape 5 - Lancement de l'analyse\n",
    "\n",
    "![Figure 6](https://github.com/IFB-ElixirFr/journeeAxeBioinfo_2020/raw/main/img/JourneeAxeStep_5.png)\n",
    "\n",
    "Pour lancer l'analyse, un script bash a été écrit. Il se décompose en 4 grandes parties :\n",
    "- Les paramètres de SLURM (l'orchestrateur de jobs disponible sur le cluster de l'IFB)\n",
    "- Le chargement des outils nécessaires à l'aide de `module load`\n",
    "- Le lancement du workflow Snakemake qui va réaliser toutes les étapes du contrôle qualité des fichiers fastq jusqu'à la génération d'une table de comptage. \n",
    "- Le lancement dans l'image singularity de l'analyse différentielle réalisé par SARtools ("
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cat runAnalysis.slurm"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![Figure 7](https://github.com/IFB-ElixirFr/journeeAxeBioinfo_2020/raw/main/img/JourneeAxeStep_6.png)\n",
    "![Figure 8](https://github.com/IFB-ElixirFr/journeeAxeBioinfo_2020/raw/main/img/JourneeAxeStep_7.png)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sbatch runAnalysis.slurm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mv slurm-* Logs/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Quelques résultats\n",
    "\n",
    "### Volcano plot"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![volcano](demo/figures/volcanoPlot.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Les gènes UP régulés**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cut -f1,18,20,21 tables/DEPLETEDvsSTANDARD.up.txt "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Les gènes DOWN régulés**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cut -f1,18,20,21 tables/DEPLETEDvsSTANDARD.down.txt "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Export\n",
    "![Figure 9](https://github.com/IFB-ElixirFr/journeeAxeBioinfo_2020/raw/main/img/JourneeAxeStep_8.png)\n",
    "\n",
    "### Exploration avec un Notebook R interactif \n",
    "\n",
    "Une petite démo rapide ? "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Merci !\n",
    "\n",
    "### La team FAIR Bioinfo \n",
    "<div id=\"banner\">\n",
    "    <div style=\"display: inline-block;\">\n",
    "        <img src=\"https://avatars2.githubusercontent.com/u/11072022?s=400&u=49843fd5dfd68775dbdd2c1e8f917a2b45e59436&v=4\" alt=\"Gildas\" style=\"border-radius: 50%;\" width=\"100px\">\n",
    "    </div>\n",
    "    <div style=\"display: inline-block;\">\n",
    "         <img src=\"https://pbs.twimg.com/profile_images/1141071921007747072/tO6YzJ0A_400x400.jpg\" alt=\"Julien\" style=\"border-radius: 50%;\" width=\"100px\">\n",
    "    </div>\n",
    "    <div style=\"display: inline-block;\">\n",
    "       <img src=\"https://migale.inrae.fr/sites/default/files/helene_chiapello.png\" alt=\"Hélène\" style=\"border-radius: 50%;\" width=\"100px\">\n",
    "    </div>\n",
    "    <div style=\"display: inline-block;\">\n",
    "       <img src=\"https://www.lri.fr///images/membre/430.jpg\" alt=\"Claire\" style=\"border-radius: 50%;\" width=\"100px\">\n",
    "    </div>\n",
    "    <div style=\"display: inline-block;\">\n",
    "       <img src=\"https://www.iscb.org/images/stories/ismbeccb2021/people/jacques.jpg\" alt=\"Jacques\" style=\"border-radius: 50%;\" width=\"100px\">\n",
    "    </div>\n",
    "    <div style=\"display: inline-block;\">\n",
    "       <img src=\"https://pbs.twimg.com/profile_images/486507449856565248/vEeEcItU_400x400.jpeg\" alt=\"Céline\" style=\"border-radius: 50%;\" width=\"100px\">\n",
    "    </div>\n",
    "    <div style=\"display: inline-block;\">\n",
    "       <img src=\"https://media-exp1.licdn.com/dms/image/C4D03AQFPPt_IGy9TGQ/profile-displayphoto-shrink_100_100/0?e=2159024400&v=beta&t=K2l9QULm8YeCVnEcai-oK47GJbVuEKe-xSdNJZI9xIU\" alt=\"Jacques\" style=\"border-radius: 50%;\" width=\"100px\">\n",
    "    </div>\n",
    "    <div style=\"display: inline-block;\">\n",
    "       <img src=\"https://www.cesgo.org/collaboration/wp-content/uploads/sites/2/avatars/200/5d9dfd4bc48e7-bpfull.jpg\" alt=\"Jacques\" style=\"border-radius: 50%;\" width=\"100px\">\n",
    "    </div>\n",
    "    \n",
    "</div>\n",
    "\n",
    "### Et aussi\n",
    "<div id=\"banner\">\n",
    "    <div style=\"display: inline-block;\">\n",
    "       <img src=\"https://0.academia-photos.com/33133954/9829661/10954491/s200_s..boulakia.jpg\" alt=\"Sarah\" style=\"border-radius: 50%;\" width=\"100px\">\n",
    "    </div>\n",
    "    <div style=\"display: inline-block;\">\n",
    "       <img src=\"https://pbs.twimg.com/profile_images/1294370956094341121/b0UJnjov_400x400.jpg\" alt=\"Gaëlle\" style=\"border-radius: 50%;\" width=\"100px\">\n",
    "    </div>\n",
    "</div>\n",
    "\n",
    "## Ressources \n",
    "\n",
    "**Cette présentation**\n",
    "- [GitHub](https://github.com/IFB-ElixirFr/journeeAxeBioinfo_2020) \n",
    "- [DockerHub](https://hub.docker.com/repository/docker/tdenecker/journee_axe_bioinfo) \n",
    "\n",
    "**FAIR_bioinfo**\n",
    "- [2019](https://github.com/thomasdenecker/FAIR_Bioinfo) \n",
    "- [2020](https://ifb-elixirfr.github.io/IFB-FAIR-bioinfo-training/)\n",
    "\n",
    "**FAIR & le cluster de l'IFB**\n",
    "- [Slurm](https://ifb-elixirfr.github.io/IFB-FAIR-bioinfo-training/assets/pdf/Session2020/04_cluster.pdf)\n",
    "- [Snakemake + Slurm](https://ifb-elixirfr.github.io/IFB-FAIR-bioinfo-training/assets/pdf/Session2020/04_tp1_snakemake.pdf)\n",
    "- [Docker/Singularity](https://ifb-elixirfr.github.io/IFB-FAIR-bioinfo-training/assets/pdf/Session2020/04_tp2_singularity.pdf)\n",
    "\n",
    "**Données**\n",
    "- [Originale](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR4026187)\n",
    "- Réduite : [![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.3997237.svg)](https://doi.org/10.5281/zenodo.3997237)\n",
    "\n",
    "**Publication**\n",
    "- Lelandais, G., Scheiber, I., Paz-Yepes, J. et al. Ostreococcus tauri is a new model green alga for studying iron metabolism in eukaryotic phytoplankton. BMC Genomics 17, 319 (2016). https://doi.org/10.1186/s12864-016-2666-6\n",
    "\n",
    "**Outils utilisés**\n",
    "- Snakemake [![Snakemake](https://img.shields.io/badge/snakemake-≥5.6.0-brightgreen.svg?style=flat)](https://snakemake.readthedocs.io)\n",
    "- [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)\n",
    "- Bowtie2 : Langmead B, Salzberg S. Fast gapped-read alignment with Bowtie 2. Nature Methods. 2012, 9:357-359.\n",
    "- SAMtools : Li H., Handsaker B., Wysoker A., Fennell T., Ruan J., Homer N., Marth G., Abecasis G., Durbin R. and 1000 Genome Project Data Processing Subgroup (2009) The Sequence alignment/map (SAM) format and SAMtools. Bioinformatics, 25, 2078-9. [PMID: 19505943](http://www.ncbi.nlm.nih.gov/pubmed/19505943)\n",
    "- featureCounts : Liao Y, Smyth GK, Shi W. featureCounts: an efficient general purpose program for assigning sequence reads to genomic features. Bioinformatics. 2014 Apr 1;30(7):923-30. doi: 10.1093/bioinformatics/btt656. Epub 2013 Nov 13. PMID: 24227677.\n",
    "- SARtools : H. Varet, L. Brillet-Guéguen, J.-Y. Coppee and M.-A. Dillies, SARTools: A DESeq2- and EdgeR-Based R Pipeline for Comprehensive Differential Analysis of RNA-Seq Data, PLoS One, 2016, doi: http://dx.doi.org/10.1371/journal.pone.0157022 \n",
    "- R : R Core Team (2020). R: A language and environment for statistical computing. R Foundation for Statistical Computing, Vienna, Austria.URL https://www.R-project.org/."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
